superset/assets/src/SqlLab/components/ExploreResultsButton.jsx

```diff
@@ -0,0 +1,167 @@
+/* eslint no-undef: 2 */
+import moment from 'moment';
+import React from 'react';
+import PropTypes from 'prop-types';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import { Alert } from 'react-bootstrap';
+import Dialog from 'react-bootstrap-dialog';
+
+import shortid from 'shortid';
+import { exportChart } from '../../explore/exploreUtils';
+import * as actions from '../actions';
+import InfoTooltipWithTrigger from '../../components/InfoTooltipWithTrigger';
+import { t } from '../../locales';
+import Button from '../../components/Button';
+
+const propTypes = {
+  actions: PropTypes.object.isRequired,
+  query: PropTypes.object,
+  errorMessage: PropTypes.string,
+  timeout: PropTypes.number,
+  database: PropTypes.object.isRequired,
+};
+const defaultProps = {
+  query: {},
+};
+
+class ExploreResultsButton extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      hints: [],
+    };
+    this.visualize = this.visualize.bind(this);
```

Comment:
nit: you can use autobind-decorators here

https://github.com/andreypopp/autobind-decorator

Reply:
We should introduce this decorator and attack the whole repo with it. In the meantime we have one less dependency and consistent-ish code (we have some binding done outside the constructors in many places...)